extensions/ql-vscode/src/common/unzip.ts

```diff
@@ -194,13 +207,21 @@ export async function unzipToDirectory(
 
     await taskRunner(
       entries.map((entry) => async () => {
+        let entryBytesExtracted = 0;
+
         const totalEntryBytesExtracted = await unzipFile(
           zipFile,
           entry,
           destinationPath,
+          (thisBytesExtracted) => {
+            entryBytesExtracted += thisBytesExtracted;
+            bytesExtracted += thisBytesExtracted;
+            reportProgress();
+          },
         );
 
-        bytesExtracted += totalEntryBytesExtracted;
+        // Should be 0, but just in case.
+        bytesExtracted += -entryBytesExtracted + totalEntryBytesExtracted;
```

Comment:
Sorry if I'm being dumb, but what is happening here? Are we removing the last reported entryBytesExtracted because they may be already counted? 🤔

Reply:
The `entryBytesExtracted` is the total size of the file that was extracted, and the `totalEntryBytesExtracted` is the size that was reported in the `entry`. This will ensure that if for some reason the reporting of the extraction is incorrect or missed some bytes, we still add the uncompressed size to ensure that at the end we get to 100%.